File: CSharpRewrite/CSharpRewrite/Assignments/Musicians/Musician.cs
using System.Threading.Channels;

namespace CSharpRewrite.Assignments.Musicians;

public class Musician
{
    public string Name { get; }
    public string Role { get; }
    public SkillLevel SkillLevel { get; }
    public ChannelReader<int> Reader { get; }

    public const int Delay = 750;

    private static readonly string[] FirstNames =
    {
        "Valerie", "Arnold", "Carlos", "Dorothy", "Keesha", "Phoebe", "Ralphie", "Tim", "Wanda",
        "Janet", "Leo", "Yuhei", "Carson"
    };

    public static readonly string[] LastNames =
    {
        "Frizzle",
        "Perlstein",
        "Ramon",
        "Ann",
        "Franklin",
        "Terese",
        "Tennelli",
        "Jamal",
        "Li",
        "Perlstein",
        "Fujioka",
        "Ito",
        "Hage"
    };

    public Musician(string role, SkillLevel skillLevel, ChannelReader<int> reader)
    {
        Name = PickName();
        Role = role;
        SkillLevel = skillLevel;
        Reader = reader;
        Console.WriteLine($"Musician {Name}, playing the {Role} entered the room.");
    }

    public bool PlaySound()
    {
        if (Reader.TryRead(out _))
        {
            Console.WriteLine($"{Name} just got back to playing in the subway.");
            return false;
        }

        switch (SkillLevel)
        {
            case SkillLevel.Good:
            {
                Console.WriteLine($"{Name} produced sound!");
                return true;
            }
            case SkillLevel.Bad:
            {
                var rng = new Random();
                var failed = rng.Next(0, 5) == 0;
                if (failed)
                {
                    Console.WriteLine($"{Name} played a false note. Uh oh.");
                    Console.WriteLine($"{Name} sucks! kicked that member out of the band! ({Role})");
                }
                else
                {
                    Console.WriteLine($"{Name} produced sound!");
                }

                return !failed;
            }
            default:
                return true;
        }
    }

    private static string PickName()
    {
        var rng = new Random();
        var firstName = FirstNames[rng.Next(0, FirstNames.Length)];
        var lastName = LastNames[rng.Next(0, LastNames.Length)];
        return $"{firstName} {lastName}";
    }
}

File: PythonRewrite/assignments/musicians/musician.py
import random
from multiprocessing.connection import Connection

from assignments.musicians.skill_level import SkillLevel

first_names = ["Valerie", "Arnold", "Carlos", "Dorothy", "Keesha", "Phoebe", "Ralphie", "Tim", "Wanda",
               "Janet", "Leo", "Yuhei", "Carson"]

last_names = ["Frizzle",
              "Perlstein",
              "Ramon",
              "Ann",
              "Franklin",
              "Terese",
              "Tennelli",
              "Jamal",
              "Li",
              "Perlstein",
              "Fujioka",
              "Ito",
              "Hage"]


def pick_name() -> str:
    first_name = random.choice(first_names)
    last_name = random.choice(last_names)
    return f'{first_name} {last_name}'


class Musician:
    def __init__(self, role: str, skill_level: SkillLevel, pipe: Connection):
        self.name = pick_name()
        self.role = role
        self.skill_level = skill_level
        self.pipe = pipe
        print(f'Musician {self.name}, playing the {self.role} entered the room.')

    def play_sound(self) -> bool:
        is_fired = self.pipe.poll(0.5)
        if is_fired:
            print(f'{self.name} just got back to playing in the subway.')
            return False

        match self.skill_level:
            case SkillLevel.GOOD:
                print(f'{self.name} produced sound!')
                return True
            case SkillLevel.BAD:
                failed = random.randint(0, 4) == 0
                if failed:
                    print(f'{self.name} played a false note. Uh oh.')
                    print(f'{self.name} sucks! kicked that member out of the band! ({self.role})')
                else:
                    print(f'{self.name} produced sound!')
                return not failed
            case _:
                return True
